homework wuchenhong

wuchenhong opened the pull request:
homework

wuchenhong commented:
У Чэньхун

IKaftanov commented on template.py:
```diff
@@ -0,0 +1,280 @@
+import requests
+
+class WeatherAPI:
```
Это не должно присутствовать в файле `template.py`

IKaftanov commented:
6 баллов.
Вам нужно комментарии к коду?

wuchenhong commented:
Уважаемый Ilya Kaftanov
Здравствуйте!


Дело в том,что я  делал с кодом "input" и сам вводил цифры из файла "cases.json",тогда все результаты было нормальны.потом я удалил "input" и превращал переменные в нормальный вид и не знаю в чём была проблема.
В процессе выполнения всё было верно. Посылаю Вам оригиальный файл с письиом.
Могу я немножно переделать  за повышение оценки?


Благодарю Вас!
С уважением
У Чэньхун

2020年10月13日 下午12:45，Ilya Kaftanov <notifications@github.com> 写道：






6 баллов.
Вам нужно комментарии к коду?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
def t1():
    a = input("please input a number")
    b = (1+a//20)*20
    return b

def t2():
    a = input("please input a string")
    b = list(a)
    b.reverse()
    c = ("".join(b))
    if len(a) == 0:
        return '""'
    else: return str(c)

def t3():
    dictionary = input("please input a dictionary")
    if dictionary=={}:
        return '""'
    else:list1 = []
    for key, value in dictionary.items():
        res = key + ": " + str(value)
        list1.append(res)
        r = "; ".join(list1)
    return r

def t4():
    s = input("please input a string")
    t = input("please input a substring")
    t = t[::-1]
    result = t in s
    return result

def t5():
    l0 = input("please input a number")
    l1 = []
    for i in l0:
        x = i.split(" ")
        e1 = int(x[0])
        e2 = int(x[1])
        e3 = int(x[2])
        e = int(x[3])
        if e == e1 * e2 * e3:
            l1.append(i)
    return l1

def t6():
    S = input("please input a string")
    skip = 0
    l=[]
    for x in reversed(S):
        if x == '#':
            skip += 1
        elif skip:skip -= 1
        else: l.append(x)
    l=l[::-1]
    a = ' '
    b=a.join(l)
    b=str(b)
    b= ''.join(b.split())
    if b=="":
        return '""'
    else: return b

def t7():
    arr = input("please input a list of strings")
    arr = list(arr)
    count_dict = {}
    for i in set(arr):
        count_dict[i] = arr.count(i)
    a = count_dict
    list1 = []
    list2 = []
    for k, v in a.items():
        if v == 1:
            list1.append(k)
            list2.append(v)
    a = dict(zip(list1, list2))
    k = list(a.keys())
    s = sum(k)
    return s

def t8():
    s = input("please input a string")
    newstr = ''.join((ch if ch in '0123456789.-e' else ' ') for ch in s)
    listOfNumbers = [float(i) for i in newstr.split()]
    maxs = int(max(listOfNumbers))
    return maxs

def t9():
    num1 = input("please input a number")
    num1 = int(num1)
    fnum1="{:0>5d}".format(num1)
    return str(fnum1)

def t10():
    x = input("please input a number")
    s = {"B", "G", "R"}
    i = 0
    if len(x) == 1:
        a = x[0]
    else:
        while i < len(x) - 1:
            if x[i] == x[i + 1]:
                a = x[i]
            else:
                s.remove(x[i])
                s.remove(x[i + 1])
                a = "".join(s)
            i += 1
            s = {"B", "G", "R"}

    return a

def t11():
    l = input("please input a list")
    l = list(l)
    a = sum(l)
    c = 0
    for i, j in enumerate(l):
        if 2 * c < a:
            c = c + j
        else:
            if 2 * c - l[i - 1] == a:
                return (i - 1)
            else:
                return("-1")

def t12():
    import re
    s = input("please input a string")
    s = str(s)
    r = re.findall(r'\d+', s)
    a = "".join(r)
    a=[a[i:i+11] for i in range(0,len(s),11)]
    a=filter(None, a)
    return a

def t13():
    a = input("please input a number")
    b = input("please input another number")
    la = list(str(a))
    lb = list(str(b))
    la = list(map(int, la))
    lb = list(map(int, lb))
    lb = list(reversed(lb))
    la = list(reversed(la))
    l = max(len(la), len(lb))
    d = []
    if len(la) > len(lb):
        lb = lb + [0] * int(len(la) - len(lb))
    else:
        la = la + [0] * int(len(lb) - len(la))
    for i in range(0, l):
        sum1 = int(la[i]) + int(lb[i])
        d.append(sum1)
    d = list(reversed(d))
    e = int(str(d).replace("[", "").replace("]", "").replace(",", "").replace(" ", ""))
    return e

def t14():
    s = input("please input a dict")
    d = {'+': 'Plus ',
         '-': 'Minus ',
         '*': 'Times ',
         '/': 'Divided By ',
         '**': 'To The Power Of ',
         '=': 'Equals ',
         '!=': 'Does Not Equal ',
         '0': 'zero',
         '1': 'one',
         '2': 'two',
         '3': 'three',
         '4': 'four',
         '5': 'five',
         '6': 'six',
         '7': 'seven',
         '8': 'eight',
         '9': 'nine'}
    l = list(d.keys())
    for i in l:
        s=s.replace(i,d[i])
    return s


def t15():
    l1 = input("please input a list")
    l1 = list(l1)
    n = len(l1)
    num1 = 0
    for i in range(n):
        for j in range(n):
            a = l1[i]
            b = a[j]

            if i == j:
                num1 += b
    l1.reverse()
    num2 = 0
    for i in range(n):
        for j in range(n):
            a = l1[i]
            b = a[j]

            if i == j:
                num2 += b
    num0 = num1 + num2
    return num0

print(t1())

IKaftanov commented:
Да, конечно.

`t2` `t3` `t5` `t10` `t11` `t12` `t14` - не проходят тесты.

По поводу бота: Я не вижу у вас двух файлов `weather_api.py` `weather_bot.py`.

Пока 8 баллов

IKaftanov commented on template.py:
```diff
@@ -0,0 +1,227 @@
+  
+def t2(string):
+    b = list(string)
+    b.reverse()
```
обращайте каждую подстроку отдельно

IKaftanov commented on template.py:
```diff
@@ -0,0 +1,227 @@
+        res = key + ": " + str(value)
+        list1.append(res)
+        r = "; ".join(list1)
+    return r
```
а если словарь пустой то возникает ошибка, переменная `r` не объявлена

IKaftanov commented on template.py:
```diff
@@ -0,0 +1,227 @@
+    l = []
+    for i in strings:
+        x = i.split(" ")
+        a = int(x[0])
```
если `x[0]` невозможно преобразовать в тип int будет ошибка

IKaftanov commented on template.py:
```diff
@@ -0,0 +1,227 @@
+                return (i - 1)
+            else:
+                return -1
+
```
а что если список пустой?

IKaftanov commented on template.py:
```diff
@@ -0,0 +1,227 @@
+
+
+def t12(lst):
+    import re
```
import ВСЕГДА пишем в начале файла

IKaftanov commented on template.py:
```diff
@@ -0,0 +1,227 @@
+    a = "".join(r)
+    a = [a[i:i + 11] for i in range(0, len(s), 11)]
+    a = filter(None, a)
+    return a
```
вы возвращаете объект фильтр.

IKaftanov commented on template.py:
```diff
@@ -0,0 +1,227 @@
+         '**': 'To The Power Of ',
+         '=': 'Equals ',
+         '!=': 'Does Not Equal ',
+         '0': 'zero',
```
Числа с большой буквы

IKaftanov commented:
13

IKaftanov commented:
на будущее, не пытайтесь рисовать
